source/Nevermore.Analyzers/NevermoreDisposeLocalDeclarationTransactionAnalyzer.cs

```diff
@@ -0,0 +1,51 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Nevermore.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class NevermoreDisposeLocalDeclarationTransactionAnalyzer : DiagnosticAnalyzer
+    {
+        readonly HashSet<string> methodsWeCareAbout = new HashSet<string> {"BeginTransaction", "BeginReadTransaction", "BeginWriteTransaction", "BeginReadTransactionAsync", "BeginWriteTransactionAsync"};
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+            context.RegisterSyntaxNodeAction(Handle, SyntaxKind.LocalDeclarationStatement);
+        }
+
+         void Handle(SyntaxNodeAnalysisContext context)
+         {
+             if (context.Node is LocalDeclarationStatementSyntax {Declaration: {Variables: var variables} localDeclaration} statement &&
+                 statement.UsingKeyword.IsKind(SyntaxKind.None))
+             {
+                 foreach (var declarator in variables)
+                 {
+                     if (declarator.Initializer is {Value: { } syntax})
+                     {
+                         var expression = (syntax as InvocationExpressionSyntax)?.Expression ??
+                                          (syntax as AwaitExpressionSyntax)?.Expression;
+
+                         if (expression is { } &&
+                             context.SemanticModel.GetSymbolInfo(expression, context.CancellationToken).Symbol is
+                                 { } symbol &&
+                             methodsWeCareAbout.Contains(symbol.Name))
+                         {
```

Comment:
`GetSymbolInfo` is expensive when I was doing some perf testing, and you can get away without it (We use it [over here](https://github.com/OctopusDeploy/RoslynAnalyzers/blob/e73d8404a9979d7f352e64f775a411379d986be1/source/RoslynAnalyzers/CaseSensitiveEnumParsingAnalyzer.cs#L46))

```suggestion
                      var expression = (syntax as AwaitExpressionSyntax)?.Expression ?? syntax;

                        if (expression is InvocationExpressionSyntax ies &&
                            ies.Expression is MemberAccessExpressionSyntax memberAccess &&
                            methodsWeCareAbout.Contains(memberAccess.Name.Identifier.Text))
                        {
```